msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackGenerator.java

```diff
@@ -109,14 +108,7 @@ public MessagePackGenerator(int features, ObjectCodec codec, OutputStream out)
         }
         messageBufferOutputHolder.set(messageBufferOutput);
 
-        if (messagePacker == null) {
-            messagePacker = MessagePack.newDefaultPacker(messageBufferOutput);
-        }
-        else {
-            messagePacker.reset(messageBufferOutput);
```

Comment:
I don't see how it can slow down the serialization process, as the creation of MessagePacker basically does nothing - just sets some fields. The thing you're right here is that I'm creating now MessagePacker for each serialization process - which means more garbage to remove later.
But because I want to provide the ability of passing config for each MessagePackFactory, I just can't hold shared state because else this feature just won't be possible.
Do you think there's better solution?